deepcell_toolbox/utils.py

```diff
@@ -130,18 +131,123 @@ def _shift_image(img, shift):
+                             str(len(data.shape))))
+
+    if len(shape) != 2:
+        raise ValueError('Shape for resize can only have length of 2, e.g. (x,y).'
+                         'Input shape has {} dimensions.'.format(str(len(shape))))
+
+    original_dtype = data.dtype
+
+    # cv2 resize is faster but does not support multi-channel data
+    # If the data is multi-channel, use skimage.transform.resize
+    channel_axis = 0 if data_format == 'channels_first' else -1
+    batch_axis = -1 if data_format == 'channels_first' else 0
+
+    # Use skimage for multichannel data
+    if data.shape[channel_axis] > 1:
+        # Adjust output shape to account for channel axis
+        if data_format == 'channels_first':
+            shape = tuple([data.shape[channel_axis]] + list(shape))
+        else:
+            shape = tuple(list(shape) + [data.shape[channel_axis]])
+
+        # linear interpolation (order 1) for image data, nearest neighbor (order 0) for labels
+        # anti_aliasing introduces spurious labels, include only for image data
+        order = 0 if labeled_image else 1
+        anti_aliasing = not labeled_image
+
+        _resize = lambda d: transform.resize(d, shape, mode='constant', preserve_range=True,
+                                             order=order, anti_aliasing=anti_aliasing)
+    # single channel image, resize with cv2
+    else:
+        shape = tuple(shape)[::-1]  # cv2 expects swapped axes.
+
+        # linear interpolation for image data, nearest neighbor for labels
+        # CV2 doesn't support ints for linear interpolation, set to float for image data
+        if labeled_image:
+            interpolation = cv2.INTER_NEAREST
+        else:
+            interpolation = cv2.INTER_LINEAR
+            data = data.astype('float32')
+
+        _resize = lambda d: np.expand_dims(cv2.resize(np.squeeze(d), shape,
+                                                      interpolation=interpolation),
+                                           axis=channel_axis)
+
+    # Check for batch dimension to loop over
+    if len(data.shape) == 4:
+        batch = []
+        for i in range(data.shape[batch_axis]):
+            d = data[i] if batch_axis == 0 else data[..., i]
+            batch.append(_resize(d))
+        resized = np.stack(batch, axis=batch_axis)
+    else:
+        resized = _resize(data)
+
+    return resized.astype(original_dtype)
+
+
+# Workaround for python2 not supporting `with tempfile.TemporaryDirectory() as`
+# These are unnecessary if not supporting python2
+
+@contextlib.contextmanager
+def cd(newdir, cleanup=lambda: True):
+    prevdir = os.getcwd()
+    os.chdir(os.path.expanduser(newdir))
+    try:
+        yield
+    finally:
+        os.chdir(prevdir)
+        cleanup()
+
+
+@contextlib.contextmanager
+def get_tempdir():
+    dirpath = tempfile.mkdtemp()
+
+    def cleanup():
+        return shutil.rmtree(dirpath)
+    with cd(dirpath, cleanup):
+        yield dirpath
+
```

Comment:
These were removed in a recent master branch commit. No need to recommit them.

Reply:
Done